filter.imp

```diff
@@ -1,13 +1,119 @@
+   { include: ["@.asm-generic/pci.h.", private, "<asm/pci.h>", "private"] },
+   { include: ["@.asm-generic/percpu.h.", private, "<asm/percpu.h>", "private"] },
+   { include: ["@.asm-generic/pgalloc.h.", private, "<asm/pgalloc.h>", "public"] },
+   { include: ["@.asm-generic/preempt.h.", private, "<asm/preempt.h>", "public"] },
+   { include: ["@.asm-generic/rwonce.h.", private, "<asm/rwonce.h>", "public"] },
+   { include: ["@.asm-generic/sections.h.", private, "<asm/sections.h>", "public"] },
+   { include: ["@.asm-generic/serial.h.", private, "<asm/serial.h>", "public"] },
+   { include: ["@.asm-generic/shmparam.h.", private, "<asm/shmparam.h>", "public"] },
+   { include: ["@.asm-generic/simd.h.", private, "<asm/simd.h>", "public"] },
+   { include: ["@.asm-generic/softirq_stack.h.", private, "<asm/softirq_stack.h>", "public"] },
+   { include: ["@.asm-generic/switch_to.h.", private, "<asm/switch_to.h>", "public"] },
+   { include: ["@.asm-generic/timex.h.", private, "<asm/timex.h>", "private"] },
+   { include: ["@.asm-generic/tlbflush.h.", private, "<asm/tlbflush.h>", "public"] },
+   { include: ["@.asm-generic/topology.h.", private, "<asm/topology.h>", "public"] },
+   { include: ["@.asm-generic/trace_clock.h.", private, "<asm/trace_clock.h>", "public"] },
+   { include: ["@.asm-generic/uaccess.h.", private, "<asm/uaccess.h>", "private"] },
    { include: ["@.asm-generic/unaligned.h.", private, "<asm/unaligned.h>", "public"] },
-   { include: ["@.asm-generic/errno.h.", private, "<linux/errno.h>", "public"] },
+   { include: ["@.asm-generic/vermagic.h.", private, "<asm/vermagic.h>", "public"] },
+   { include: ["@.asm-generic/vga.h.", private, "<asm/vga.h>", "public"] },
+   { include: ["@.asm-generic/word-at-a-time.h.", private, "<asm/word-at-a-time.h>", "public"] },
+   { include: ["@.asm-generic/xor.h.", private, "<asm/xor.h>", "public"] },
+
+
+   # The below headers were added by hand due to errors being found when the private headers were included
+   { include: ["<asm/bitsperlong.h>", private, "<linux/bits.h>", "public"] },
+   { include: ["<asm/errno.h>", private, "<linux/errno.h>", "public"] },
+   { include: ["<asm/resource.h>", private, "<linux/resource.h>", "public"] },
+   { include: ["@.asm/signal-defs.h.", private, "<linux/signal.h>", "public"] },
+   { include: ["<asm/signal.h>", private, "<linux/signal.h>", "public"] },
+   { include: ["@.asm-generic/bitops*.", private, "<linux/bitops.h>", "public"] },
+   { include: ["<asm/bug.h>", private, "<linux/bug.h>", "public"] },
+   { include: ["<asm/delay.h>", private, "<linux/delay.h>", "public"] },
+   { include: ["<asm/io.h>", private, "<linux/io.h>", "public"] },
+   { include: ["@.asm/kprobes.h.", private, "<linux/kprobes.h>", "public"] },
+   { include: ["<asm/module.h>", private, "<linux/module.h>", "public"] },
+   { include: ["<asm/pci.h>", private, "<linux/pci.h>", "public"] },
+   { include: ["<asm/percpu.h>", private, "<linux/percpu.h>", "public"] },
```

Comment:
these should all be consistent in their use of `@` in the regex.

Reply:
I feel like this block of rules should either always use the regex (`@`) or never; but not BOTH interspersed throughout the file.

For instance, what if someone has `#include "asm/percpu.h"`?  That wont match the rule for `<asm/percpu.h>` so it won't be replaced with `<linux/percpu.h>`.